pkg/apis/common/v1alpha1/object_interfaces.go

```diff
@@ -39,6 +39,9 @@ type Reconcilable interface {
 
 	// GetStatusManager returns a manager for the component's status.
 	GetStatusManager() *StatusManager
+
+	// GetAdapterOverrides returns the adapter overrides.
+	GetAdapterOverrides() *AdapterOverrides
```

Comment:
I was about to suggest that we turn this into its own interface, so that we take a Go-_ish_ behaviour-driven approach and allow components to selectively implement overrides or not.
But I see you already implemented the method everywhere (crazy you!) so my suggestion suddenly becomes less relevant.

Still, if we consider that a type can be reconcilable _without_ having to override adapter settings, it might make sense to think about this as an additional `AdapterConfigurable` interface-driven behavior of some sort. Even if we want to encourage all integrations to support it at the end of the day, there might be exceptions.
The very important difference with `Reconcilable` is that, if a component doesn't implement an optional interface, things can **work** nonetheless (which is the case here).

An example of this is `EventSource` below, which can _optionally_ be implemented to expose some information about CE source and types.  In practice, all `source` types should implement it, right?
Nope! We already found a case of Source that will _not_ implement it with @odacremolbap's new `CloudEventsSource`.